packages/server/src/adapters/bun/createBunHttpHandler.ts

```diff
@@ -0,0 +1,24 @@
+import type {Server} from "bun";
+import {fetchRequestHandler} from "../fetch";
+import type { AnyRouter, inferRouterContext } from "../../core";
+import type { HTTPBaseHandlerOptions } from "../../http";
+
+export type BunHttpHandlerOptions<TRouter extends AnyRouter> = HTTPBaseHandlerOptions<TRouter, Request> & {
+    endpoint: string;
+    createContext?: (opts: { req: Request }) => inferRouterContext<TRouter> | Promise<inferRouterContext<TRouter>>;
+};
+
+export function createBunHttpHandler<TRouter extends AnyRouter>(opts: BunHttpHandlerOptions<TRouter>) {
+    return (request: Request, server: Server) => {
+        if (
+            server.upgrade(request, {data: {req: request}})
+        ) {
+            return new Response(null, {status: 101});
```

Comment:
based on buns docs, I believe it automatically returns 101 if the connection was successful:

https://bun.sh/guides/websocket/upgrade

From your point of view do you think this if statement is required?

Reply:
By my point of view this statement is required by type signature specified [here](https://github.com/oven-sh/bun/blob/b6f60680f699f63c37e416b54ef1ebd3a14aaff5/packages/bun-types/bun.d.ts#L2105).